### PR TITLE
ENH: eog/ecg n comp

### PR DESCRIPTION
**Pull request by dengemann**
@agramfort @hoechenberger here is the first portion

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by hoechenberger** [comment]
We should add a test case. Will do.

**Comment by hoechenberger** [comment]
I figured that the parameters `average_ecg_projs` and `average_eog_projs` and their respective descriptions:
```
 Whether to average [ECG,EOG] proejction vectors or not.
```
were a bit misleading, because according to the MNE docs, the `average` parameter determines whether the SSP vectors are to be calculated on epochs (`average=False`) or evoked data (`average=True`).

I, therefore, have renamed the parameters and adjusted their docstrings:

```python
ecg_proj_from: Literal['epochs', 'evoked'] = 'evoked'
"""
Whether to calculate the ECG proejction vectors based on the evoked ECG signal
 (i.e., the averaged ECG epochs), or on the ECG epochs.
"""
```

Okay with you, @agramfort and @dengemann?

`ds000248` test passes locally for me.

File: config.py [hunk 1]
```diff
@@ -903,6 +903,73 @@
...
+"""
+
+n_proj_ecg: Dict[str, float] = dict(n_mag=1, n_grad=1, n_eeg=1)
+"""
+Number of SSP vectors to create for ECG artifacts for each channel type.
+"""
+
+ecg_proj_from: Literal['epochs', 'evoked'] = 'evoked'
+"""
+Whether to calculate the ECG proejction vectors based on the evoked ECG signal
```
Comment (agramfort) [inline config.py:922]:
```suggestion
Whether to calculate the ECG projection vectors based on the evoked ECG signal
```

File: config.py [hunk 2]
```diff
@@ -903,6 +903,73 @@
...
+(i.e., the averaged ECG epochs), or on the ECG epochs.
+"""
+
+eog_proj_from: Literal['epochs', 'evoked'] = 'evoked'
+"""
+Whether to calculate the EOG proejction vectors based on the evoked EOG signal
```
Comment (agramfort) [inline config.py:928]:
```suggestion
Whether to calculate the EOG projection vectors based on the evoked EOG signal
```

File: config.py [hunk 3]
```diff
@@ -1616,6 +1683,15 @@ def check_baseline(
 check_baseline(baseline=baseline, epochs_tmin=epochs_tmin,
                epochs_tmax=epochs_tmax)
 
+# SSP projectors baswed on evoked or epochs
+if ecg_proj_from not in ('evoked', 'epochs'):
+    raise ValueError(f'ecg_proj_from must be one of: evoked, epochs; but got: '
```
Comment (agramfort) [inline config.py:1688]:
```suggestion
    raise ValueError(f'ecg_proj_from must be one of: "evoked", "epochs"; but got: '
```

File: config.py [hunk 4]
```diff
@@ -1616,6 +1683,15 @@ def check_baseline(
...
+                     f'{ecg_proj_from} instead.')
+
+if eog_proj_from not in ('evoked', 'epochs'):
+    raise ValueError(f'eog_proj_from must be one of: evoked, epochs; but got: '
```
Comment (agramfort) [inline config.py:1692]:
```suggestion
    raise ValueError(f'eog_proj_from must be one of: "evoked", "epochs"; but got: '
```

File: tests/configs/config_ds000248_rest.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+"""
+MNE Sample Data
+"""
```
Comment (agramfort) [inline tests/configs/config_ds000248_rest.py:3]: this file should not be in this PR

Comment (hoechenberger) [reply tests/configs/config_ds000248_rest.py]: Thanks, removed

File: tests/configs/config_ds000248.py [hunk 6]
```diff
@@ -23,6 +23,10 @@
 noise_cov = 'emptyroom'
 
 spatial_filter = 'ssp'
+n_proj_eog = dict(n_mag=1, n_grad=1, n_eeg=1)
+n_proj_ecg = dict(n_mag=1, n_grad=1, n_eeg=0)
+ecg_proj_from = 'evoked'
+eog_proj_from = 'epochs'
```
Comment (dengemann) [inline tests/configs/config_ds000248.py:29]: I find this very unintuitive @hoechenberger @agramfort -- if we average or not does not depend on on whether it's epochs or evoked does it? I would find a simple average parameter more intuitive, matching the MNE API.

**Comment by dengemann** [comment]
> I figured that the parameters `average_ecg_projs` and `average_eog_projs` and their respective descriptions:
> 
> ```
>  Whether to average [ECG,EOG] proejction vectors or not.
> ```
> 
> were a bit misleading, because according to the MNE docs, the `average` parameter determines whether the SSP vectors are to be calculated on epochs (`average=False`) or evoked data (`average=True`).
> 
> I, therefore, have renamed the parameters and adjusted their docstrings:
> 
> ```python
> ecg_proj_from: Literal['epochs', 'evoked'] = 'evoked'
> """
> Whether to calculate the ECG proejction vectors based on the evoked ECG signal
>  (i.e., the averaged ECG epochs), or on the ECG epochs.
> """
> ```
> 
> Okay with you, @agramfort and @dengemann?
> 
> `ds000248` test passes locally for me.

I think I'm -1 on the naming, it is confusing at multiple levels. I would not call link averaging in this context to epochs VS evoked.

File: tests/configs/config_ds000248.py (hunk 6 above)
Comment (agramfort) [reply tests/configs/config_ds000248.py:29]: @dengemann you would prefer ecg_proj_epochs_average= bool ?

Personally I prefer typing True or False but no strong feeling. A book is also closer to MNE API.

**Comment by hoechenberger** [comment]
@dengemann 
I've addressed this in cabb799, better now?

File: tests/configs/config_ds000248.py (hunk 6 above)
Comment (dengemann) [reply tests/configs/config_ds000248.py:29]: yes I would. For the same reasons, hence, my proposal.

**Comment by agramfort** [comment]
honestly I would prefer 

eog_proj_from_average = True | False
ecg_proj_from_average = True | False

**Comment by hoechenberger** [comment]
Thanks @dengemann!

**Pull request merged**